modules/control/common/mrac_controller.cc

```diff
@@ -262,6 +265,45 @@ bool MracController::CheckLyapunovPD(const Matrix matrix_a,
           llt_matrix_q.info() != Eigen::NumericalIssue);
 }
 
+void MracController::EstimateInitialGains(const LatencyParam &latency_param) {
+  const double rise_time_estimate =
+      latency_param.dead_time() + latency_param.rise_time();
+  const double settling_time_estimate =
+      latency_param.dead_time() + latency_param.settling_time();
+  Matrix matrix_a_estimate = Matrix::Zero(model_order_, model_order_);
+  Matrix matrix_b_estimate = Matrix::Zero(model_order_, 1);
+  if (model_order_ == 1 &&
+      (rise_time_estimate >= ts_ || settling_time_estimate >= ts_)) {
+    const double tau_estimate = (rise_time_estimate >= ts_)
+                                    ? rise_time_estimate / 2.2
+                                    : settling_time_estimate / 4.0;
+    matrix_a_estimate(0, 0) = -1.0 / tau_estimate;
+    matrix_b_estimate(0, 0) = 1.0 / tau_estimate;
+  } else if (model_order_ == 2 &&
+             (rise_time_estimate >= ts_ && settling_time_estimate >= ts_)) {
+    const double wn_estimate = 1.8 / rise_time_estimate;
+    const double zeta_estimate =
+        4.6 / (rise_time_estimate * settling_time_estimate);
+    matrix_a_estimate(0, 1) = 1.0;
+    matrix_a_estimate(1, 0) = -wn_estimate * wn_estimate;
+    matrix_a_estimate(1, 1) = -2 * zeta_estimate * wn_estimate;
+    matrix_b_estimate(1, 0) = wn_estimate * wn_estimate;
+  } else {
+    AWARN << "No pre-known actuation dynamics; the initial states of the "
+             "adaptive gains are set as zeros";
+    return;
+  }
+  gain_state_adaption_.col(1) =
```

Comment:
You are just calculating psedo inverse here, so try to re-use the function in `modules/common` https://github.com/ApolloAuto/apollo/blob/9e3c56a7342800a5e06aac46f9acf905071f03bc/modules/common/math/matrix_operations.h#L51

Reply:
re-use the common/math function per comment. Thanks.